src/components/newsletter-form.jsx

```diff
@@ -0,0 +1,123 @@
+"use strict";
+
+import $ from "jquery";
+import React from "react";
+import { Row, Col, Input, ButtonInput, ListGroup, ListGroupItem } from "react-bootstrap";
+
+export default React.createClass({
+  displayName: "NewsletterForm",
+
+  getEmailRegex() {
+      return /^[a-zA-Z0-9.!#$%&’*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]{2,})+$/;
+  },
+
+  getInitialState() {
+    return {
+      subscribed: false,
+      hasErrored: false,
+      newsletterEmail: ""
+    };
+  },
+
+  onEmailChange(event) {
+    this.setState({
+      newsletterEmail: event.currentTarget.value
+    });
+  },
+
+  validationState() {
+    if (!this.state.newsletterEmail.length) {
+      return null;
+    }
+    if (this.getEmailRegex().exec(this.state.newsletterEmail)) {
```

Comment:
Schimba metoda `getEmailRegex` cu o chestie care sa includa rulatul regex-ului. Denumeste-o `checkEmailIsCorrect` sau ceva asemanator si fa replace si in metoda `onSubmit`.